esx_service/vmdk_ops.py

```diff
@@ -837,25 +838,31 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     except ValidationError as ex:
         return err(str(ex))
 
-    if datastore and not vmdk_utils.validate_datastore(datastore):
+    if not datastore:
+        datastore = default_datastore
+
+    # Its possible the datastore is None, which can happen if datastores on the
+    # host could not be identified or initialized with the dockvols path.
+    if not datastore:
```

Comment:
I had kept both cases together but the error message is indicating "default_datastore = None" (which can happen if there were errors in identifying datastores on the host and hence missing in the cache). The user doesn't know how the default_datastore is identified, what it is and why its none. I can remove default_datastore and make a single error message perhaps.